src/init.cpp

```diff
@@ -1971,14 +1968,18 @@ bool AppInitMain(const CoreContext& context, NodeContext& node, interfaces::Bloc
                     node.llmq_ctx->Stop();
                 }
                 node.llmq_ctx.reset();
-                node.llmq_ctx = std::make_unique<LLMQContext>(chainman.ActiveChainstate(), *node.connman, *node.dmnman, *node.evodb, *node.mn_metaman, *node.mnhf_manager, *node.sporkman,
-                                                              *node.mempool, node.mn_activeman.get(), *node.mn_sync, node.peerman, /* unit_tests = */ false, /* wipe = */ fReset || fReindexChainState);
+                node.llmq_ctx = std::make_unique<LLMQContext>(chainman.ActiveChainstate(), *node.connman, *node.dmnman, *node.evodb, *node.mn_metaman, *node.sporkman, *node.mempool,
+                                                              node.mn_activeman.get(), *node.mn_sync, node.mnhf_manager, node.peerman, /* unit_tests = */ false, /* wipe = */ fReset || fReindexChainState);
                 // Have to start it early to let VerifyDB check ChainLock signatures in coinbase
                 node.llmq_ctx->Start();
 
+                node.mnhf_manager.reset();
```

Comment:
why that is done? `mnhf_manager` is needed to determine if soft-forks are started and this manager if it is not used right now widely, but it means to be initialized close to `CChainState` soon as possible, because everything that uses chain may also be interested by status of soft-fork by BIP9. Everything else, such as QuorumManager, Metaman, mn_sync, peerman, quorum-instant-send-lock....

`Mnhf_manager` should be initilized early as possible to be sure when `CChainState` (and its helpers) are initialized you already have a knowledge about soft-forks

Reply:
```
const auto quorum = llmq::quorumManager->GetQuorum(llmqType, quorumHash);
```

due to that. Hm, is there any way to resolve this deglob without reversing initialization order?

Reply:
I initially took the approach of having {`Process`, `Undo`}`Block` accept `llmq::CQuorumManager`, which seemed sensible since it would be called through `CSpecialTxProcessor`, which has access to `llmq::CQuorumManager`. Unfortunately, `ProcessBlock` is also used by `GetForBlock`, which is used by `GetSignalsStage` and `AddSignal`, the former being used in `VersionBitsConditionChecker`.

Since threading that needle would be unpleasant, the less unpleasant means would be to give `CMNHFManager` knowledge of `llmq::CQuorumManager`. That's why I took the approach that I did.

Alternatively, we can do what we did for `CTxMemPool` and leave certain functionality _unavailable_ unless you call `ConnectManagers`, which then let's you use the desired functionality. Since that was quite easy to do for `CTxMemPool` (most of the ProTx validation code was quite easy to segment), it was done there, I wasn't sure about that here. 

If `CMNHFManager` could be given a read-only mode (which only needs `CEvoDB`) and full mode (`llmq::CQuorumManager`), this approach would be plausible.

Reply:
Refactored the code to now have (`Dis`)`connectManagers`, `llmq::CQuorumManager` is used only {`Process`, `Undo`}`Block`, appropriate comments and `assert`s have been added.